### ISSUE TITLE
Help with choosing appropriate widgets for project

### ISSUE THREAD

**Issue by jeffwright13** [ceccopierangiolieugenio/pyTermTk#32]
[Pytest-fold](https://github.com/jeffwright13/pytest-fold) is a Pytest plugin that attempts to display results of a Pytest run in a way that is easier to visualize than Pytest's "dump it all to console" output. The idea is to display Pass/Fail (and maybe other) results on the console in a "folded" manner, where only the title is visible until it is clicked; at which time the full details of that test are "unfolded" onto the console for viewing. If that test is clicked again, it folds back up. You can get an idea of what I mean by looking at the 2nd and 3rd movies in the [README](https://github.com/jeffwright13/pytest-fold/blob/pytermtk/README.md). I have already implemented two simple versions of the clickable 'fold' feature - one in Asciimatics, and one in Textual. Now I am trying to do a TUI version in PyTermTk, and would like some advice on what widgets might be best for implementing this idea.

Basically, I need a widget that displays a line of text (the test titles), allows me to click on it, and then displays full test detail information. In the two movies I mentioned above, you can see two different ways I've done it - (a) a list of test titles stacked vertically that can be clicked, and which open/close in place; (b) a tree that lists test titles as nodes, and then displays the test details in a separate text box.

I already have a simple TUI built up in PyTermTk, in which the various output sections from Pytest are displayed in tabbed text boxes. Because I am not super-familiar with the Tk way of doing things, I thought I would ask an expert for a suggestion on how to do this. :-) Any ideas?

**Comment by ceccopierangiolieugenio** [comment]
Mmmm, this is quite tricky,
I mean, 
you can try to use a combination of areawidgets, textedit or tree or buttons that pop in/out textareas and so on.
I think that at the moment, the easiest thing to do is to write your own widget so you can have full control on the what you want to achieve, I did the same for the viewer widget in [tlogg](https://github.com/ceccopierangiolieugenio/tlogg)
I can help you to create a custom widget if you need, it should be straight forward to adapt the text edit and add the mouse interaction to expand/reduce the different sections with/without wrap.
If you provide me an example output I can give it a try.

**Comment by jeffwright13** [comment]
Thanks for your contribution to my project. Closing.
